### DISCUSSION TITLE
Split large models across your GPU(s), CPU, and disk.

### DISCUSSION THREAD

**Discussion by RedDragonGecko** [oobabooga/text-generation-webui#594] [Q&A] [answered]
Does this mean it's possible to split models between the vram of two or more gpus? I haven't found any information on if it's supported outside this statement in the readme.

**Comment by BetaDoggo** [comment, marked as answer]
Yep, you can do it using the `--auto-devices` arg. You will also likely need to use `--gpu-memory` to specify how much memory to use per gpu.

**Reply by RedDragonGecko** [reply]
Thanks much. Follow up question. is a nv-link (for nvidia) required/recommended?

**Reply by BetaDoggo** [reply]
It's not required and I don't know if it's supported currently.

**Reply by RedDragonGecko** [reply]
Thanks again. I'm putting together a home server I can dedicate to ai and I'm trying to cut costs where possible.

**Reply by H2RnD** [reply]
How might I split it into ram and disk space?

**Comment by quarterturn** [comment]
Unfortunately this isn't working for me with GPTQ-for-LLaMA. I launch with ``` python server.py --model llama-30b-4bit-128g  --auto-devices --gpu-memory 16 16  --chat --listen --wbits 4 --groupsize 128 ``` but get a ```torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 22.00 MiB (GPU 0; 15.89 GiB total capacity; 15.12 GiB already allocated; 18.12 MiB free; 15.30 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF```

**Reply by BetaDoggo** [reply]
From what I've heard, when using multiple devices you have to set the gpu memory to a few GB less than its actual capacity. Try setting them to 12 or 14.

**Reply by quarterturn** [reply]
Thanks. This is indeed true, and it now works for me.

**Reply by cduk** [reply]
Hi. Could you please give an indication of performance? I need to run bigger models, but wonder if I can get away with buying another smaller card instead of plumping for a model with 24GB.
